tests/js/client/shell/shell-global-shard-statistics-cluster.js

```diff
@@ -0,0 +1,259 @@
+// / See the License for the specific language governing permissions and
+// / limitations under the License.
+// /
+// / Copyright holder is triAGENS GmbH, Cologne, Germany
+// /
+// / @author Jan Steemann
+// //////////////////////////////////////////////////////////////////////////////
+
+let jsunity = require('jsunity');
+let arangodb = require('@arangodb');
+let db = arangodb.db;
+let { getServersByType } = require('@arangodb/test-helper');
+
+function shardStatisticsSuite() {
+  'use strict';
+
+  const cn = "UnitTestsDatabase";
+
+  let fetchStats = function (append = '') {
+    let old = db._name();
+    db._useDatabase("_system");
+    let result = arango.GET("/_admin/cluster/shardStatistics" + append).result;
+    db._useDatabase(old);
+    return result;
+  };
+
+  let cleanCollections = function () {
+    let old = db._name();
+    db._useDatabase(cn);
+    for (let i = 0; i < 5; ++i) {
+      db._drop(cn + i);
+    }
+    db._useDatabase(old);
+  };
+
+  return {
+    setUpAll : function () {
+      db._useDatabase("_system");
+      try {
+        db._dropDatabase(cn);
+      } catch (err) {}
+      db._createDatabase(cn);
+    },
+    
+    tearDownAll : function () {
+      db._useDatabase("_system");
+      try {
+        db._dropDatabase(cn);
+      } catch (err) {}
+    },
+    
+    testShardStatisticsNewDatabase: function () {
+      let baseValues = fetchStats();
+
+      try {
+        db._useDatabase("_system");
+        db._createDatabase(cn + "xxxx");
+
+        let newValues = fetchStats();
+
+        assertEqual(baseValues.databases + 1, newValues.databases);
+        assertTrue(baseValues.collections < newValues.collections);
+        assertTrue(baseValues.shards < newValues.shards);
+        assertTrue(baseValues.leaders < newValues.leaders);
+        assertTrue(baseValues.followers < newValues.followers);
+        assertTrue(baseValues.realLeaders < newValues.realLeaders);
+      } finally {
+        db._useDatabase("_system");
+        db._dropDatabase(cn + "xxxx");
+      }
+    },
+    
+    testShardStatisticsPerDatabase: function () {
+      let baseValues = fetchStats();
+
+      try {
+        db._useDatabase("_system");
+        for (let i = 0; i < 5; ++i) {
+          db._createDatabase(cn + "test" + i);
+        }
+
+        let newValues = fetchStats();
+
+        assertEqual(baseValues.databases + 5, newValues.databases);
+        assertTrue(baseValues.collections < newValues.collections);
+        assertTrue(baseValues.shards < newValues.shards);
+        assertTrue(baseValues.leaders < newValues.leaders);
+        assertTrue(baseValues.followers < newValues.followers);
+        assertTrue(baseValues.realLeaders < newValues.realLeaders);
+
+        newValues = fetchStats("?details=true");
+        let keys = Object.keys(newValues);
+
+        let expected = ["_system", cn];
+        for (let i = 0; i < 5; ++i) {
+          expected.push(cn + "test" + i);
```

Comment:
You could put the expected logic to the above loop in Line: 95, but that is micro-optimization. Your call.

Reply:
Ok, I did that!